clang/lib/CodeGen/Targets/AArch64.cpp

```diff
@@ -812,6 +819,23 @@ Address AArch64ABIInfo::EmitMSVAArg(CodeGenFunction &CGF, Address VAListAddr,
                           /*allowHigherAlign*/ false);
 }
 
+void AArch64TargetCodeGenInfo::checkFunctionCallABI(
+    CodeGenModule &CGM, SourceLocation CallLoc, const FunctionDecl *Caller,
+    const FunctionDecl *Callee, const CallArgList &Args) const {
+  if (!Callee->hasAttr<AlwaysInlineAttr>())
+    return;
+
+  auto CalleeIsStreaming =
+      Sema::getArmStreamingFnType(Callee) == Sema::ArmStreaming;
+  auto CallerIsStreaming =
+      Sema::getArmStreamingFnType(Caller) == Sema::ArmStreaming;
+
+  if (CalleeIsStreaming && !CallerIsStreaming)
```

Comment:
What about mismatching the other way?
```suggestion
  if (CalleeIsStreaming != CallerIsStreaming)
```

Reply:
That makes sense, thank you. I also realised that it'll need to check if the caller is streaming_compatible but the callee isn't, so I'll add that too.

Reply:
Done